### PR TITLE
Adding function to list Github repositories

### PR DESCRIPTION
**Pull request by jossday**
New function added in hybridauth/Hybrid/Providers/Github.php
This function allows to list repositories of the user who connects
himself through hybridauth. Just use it by calling getUserRepos().

### CODE REVIEW

**Comment by nazar-pc** [comment]
1. This is Third party provider, why did you copy it into core providers?
2. I feel strongly against addition of new method - this is just API call, we can add multitude of them for each provider, but this is not an intention of this project

File: hybridauth/Hybrid/Providers/GitHub.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+				throw new Exception( "User email request failed! {$this->providerId} returned an error: $e", 6 );
+			}
+		}
+
+		return $this->user->profile;
+	}
+	/**
+	*
+	*/
+	function getUserContacts() {
```
Comment (StorytellerCZ) [inline hybridauth/Hybrid/Providers/GitHub.php]: Missing comment here.

**Comment by StorytellerCZ** [comment]
As @nazar-pc said.

**Comment by jossday** [comment]
Ok, sorry I only wanted to help people who could look for this.

**Comment by StorytellerCZ** [comment]
@jossday No harm done :) Feel free to add it to the Github provider in [additional providers](https://github.com/hybridauth/hybridauth/tree/master/additional-providers/hybridauth-github/Providers)

**Comment by nazar-pc** [comment]
Even better, create own custom provider for your project by extending class of GitHub provider.

As far as I remember this library abstracts some things across all providers, but many other things are available in individual APIs and my opinion is hat those should not be added to providers itself, because we'll end up with infinitely bloated and fragile classes. For individual purposes this might be useful though.

**Pull request closed without merging**